lib/pbench/test/unit/server/query_apis/commons.py

```diff
@@ -282,16 +291,22 @@ def test_http_exception(
         user_ok,
         find_template,
         pbench_token,
+        provide_metadata,
     ):
         """
         Check that an exception in calling Elasticsearch is reported correctly.
         """
         if not self.elastic_endpoint:
             pytest.skip("skipping " + self.test_http_exception.__name__)
 
-        index = self.build_index(
-            server_config, self.date_range(self.payload["start"], self.payload["end"])
-        )
+        if self.payload.get("start") and self.payload.get("end"):
```

Comment:
Although it may not matter in the cases we've seen, I think we should be more specific than this. Maybe it should be based on the class *schema* containing start/end? Or maybe it should be explicit in your test class (e.g., as a constructor parameter).

Reply:
Yeah...this feels weird:  if we have exactly one of start or end, we build the index from metadata?  It's not obvious how that follows logically...which makes me inclined to think that this test is wrong.  (Should the test be if we have _neither_ of them, then we use the metadata?)

Reply:
Right, so I could use class schema to determine the if the start and end are part of required keys and then determine which function to use based on that, like following:
```
parameter_items = self.cls_obj.schema.parameters.items()
required_keys = [
    key for key, parameter in parameter_items if parameter.required
]
if all(x in required_keys for x in ["start", "end"]):
    index = self.build_index(
        server_config,
        self.date_range(self.payload["start"], self.payload["end"]),
    )
elif any(x in required_keys for x in ["start", "end"]):
    pytest.skip("skipping " + self.test_http_error.__name__)
else:
    index = self.build_index_from_metadata()
```
Or I could just use another parameter in the constructor like this:
```
use_index_from_metadata: bool = False
```
and if a class wants to use the index from metadata it can set it to true. So I guess 2nd option seems simpler to me (Unless you guys think first is more dynamic)